storage/src/adb/any/variable/mod.rs

```diff
@@ -325,15 +326,17 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Codec, H: CHasher, T: Translato
                 }
             }
         }
-        if end_loc < self.log_size {
+
+        // Rewind the operations log if necessary.
+        if let Some((end_loc, end_offset)) = after_last_commit {
+            assert!(!uncommitted_ops.is_empty());
             warn!(
                 op_count = uncommitted_ops.len(),
                 log_size = end_loc,
+                end_offset,
                 "rewinding over uncommitted operations at end of log"
             );
-            // We use saturating_sub below for the case where end_loc == 0, which happens when there
-            // are no committed operations at all remaining.
-            let prune_to_section = end_loc.saturating_sub(1) / self.log_items_per_section;
+            let prune_to_section = end_loc / self.log_items_per_section;
             self.log
                 .rewind_to_offset(prune_to_section, end_offset)
```

Comment:
In `freezer`, I `rewind` (based on `size`) instead of `offset` because it is "more precise" (doesn't leave empty space at the end of the log although harmless):
* https://github.com/commonwarexyz/monorepo/blob/d49fa191f006e6fd564748fd95e8c6c0bc6951fc/storage/src/journal/variable.rs#L659-L705
* https://github.com/commonwarexyz/monorepo/blob/d49fa191f006e6fd564748fd95e8c6c0bc6951fc/storage/src/freezer/storage.rs#L579-L580

IMO it would be more "correct" to do this (given we have `size` above already).

Reply:
That being said, all the other interactions are definitely over `offset` so feels weird to be jumping between (I guess this is a bit like the `position` vs `location` in MMR).